src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -1249,6 +1345,57 @@ std::string ImageReplayer<I>::to_string(const State state) {
 }
 
 template <typename I>
+void ImageReplayer<I>::resync_image(Context *on_finish) {
+  dout(20) << dendl;
+
+  bool is_stopped = false;
+  {
+    Mutex::Locker l(m_lock);
+
+    m_stopping_for_resync = true;
+    if (m_state == STATE_STOPPED) {
```

Comment:
Is this just to service the asok method?  If so, can we just hook the asok method to librbd's request resync method?  If replay has stopped, it will be restarted every 30 seconds so we should detect the request.